File: test_regress/t/t_dist_fixme.pl
#!/usr/bin/env perl
if (!$::Driver) { use FindBin; exec("$FindBin::Bin/bootstrap.pl", @ARGV, $0); die; }
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2003 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

use IO::File;
use File::Spec::Functions 'catfile';

scenarios(dist => 1);

my $root = "..";
my $Debug;

if (!-r "$root/.git") {
    skip("Not in a git repository");
} else {
    ### Must trim output before and after our file list
    my $files = `cd $root && git ls-files --exclude-standard`;
    print "ST $files\n" if $Debug;
    my %names;

    $files =~ s/\s+/ /g;
    my @batch;
    my $n = 0;
    my $re = qr/(FIX[M]E|BO[Z]O)/;
    foreach my $file (split /\s+/, $files) {
        my $filename = catfile($root, $file);
        next if !-r $filename;
        my $wholefile = file_contents($filename);
        if ($wholefile =~ /$re/) {
            $names{$file} = 1;
        }
    }
    if (scalar(%names) >= 1) {
        error("Files with FIX" . "MEs: ", join(' ', sort keys %names));
    }
}

ok(1);
1;

File: test_regress/t/t_dist_fixme.py
#!/usr/bin/env python3
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2024 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

import vltest_bootstrap

test.scenarios('dist')

if not os.path.exists(test.root + "/.git"):
    test.skip("Not in a git repository")

### Must trim output before and after our file list
files = test.run_capture("cd " + test.root + " && git ls-files --exclude-standard")
if test.verbose:
    print("ST " + files)

names = {}
files = re.sub(r'\s+', ' ', files)

regex = r'(FIX[M]E|BO[Z]O)'
for filename in files.split():
    if "test_regress/t/t_0_uvm_dpi/" in filename:  # Standard file - can't change it
        continue
    if re.search(regex, filename):
        names[filename] = True
    filename = os.path.join(test.root, filename)
    if not os.path.exists(filename):
        continue
    wholefile = test.file_contents(filename)
    if re.search(regex, wholefile):
        names[filename] = True

if len(names):
    test.error("Files with FIX" + "MEs: " + ' '.join(sorted(names.keys())))

test.passes()
